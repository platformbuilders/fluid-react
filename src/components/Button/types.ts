import { ReactNode } from 'react';
import {
  ButtonColorType,
  ButtonVariants,
  TypographyVariants,
} from '@platformbuilders/theme-toolkit';
import RadixIcon from '@radix-ui/react-icons';

export type ButtonProps = {
  style?: any;
  type?: 'button' | 'submit' | 'reset';
  loading?: boolean;
  $fullWidth?: boolean;
  variant?: ButtonVariants;
  typographyVariant?: TypographyVariants;
  children?: string | ReactNode;
  leftIconName?: keyof typeof RadixIcon;
  rightIconName?: keyof typeof RadixIcon;
  colorVariant?: ButtonColorType;
  size?: 'normal' | 'medium' | 'large';
} & TouchableType;

export type TouchableType = {
  id?: string;
  accessibility: string;
  testID?: string;
  disabled?: boolean;
  onPress?: (param: any) => void;
};
