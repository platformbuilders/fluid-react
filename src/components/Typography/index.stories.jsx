import React from 'react';
import { storiesOf } from '@storybook/react';
import { withKnobs, select } from '@storybook/addon-knobs';
import markdown from './typography.md';
import Typography from './index';

const optionsVariant = [
  'h1',
  'h2',
  'h3',
  'h4',
  'h5',
  'h6',
  'subtitle1',
  'subtitle2',
  'body1',
  'body2',
  'button',
  'caption',
  'overline',
]
storiesOf('Typography', module)
  .addDecorator(withKnobs)
  .addParameters({
    readme: {
      content: markdown,
    },
    viewport: {
      defaultViewport: '',
    },
  })
  .add('Example', () => (
    <Typography variant={select('variant', optionsVariant, 'h1')}>
      Typography Example
    </Typography>
  ));
