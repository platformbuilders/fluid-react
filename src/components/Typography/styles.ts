import styled from 'styled-components/native';
import { getTheme } from '../../utils/helpers';

const defaultColor = getTheme('primary.contrast');

export const Text = styled.Text`
  color: ${defaultColor};
`;
