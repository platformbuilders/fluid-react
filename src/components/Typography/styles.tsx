import styled from 'styled-components';
import { Typography } from '@material-ui/core';

export const CustomTypography = styled(Typography)``;
