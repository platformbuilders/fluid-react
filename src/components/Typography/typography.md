
# Typography Example usages 

### Source Code
<!-- SOURCE -->

```html
<Typography as="h1" variant="max">
  Logout
</Typography>
<Title forwardedAs="h1" variant="max">
  {title}
</Title>
<Title2>Welcome</Title2>
```

```ts
export const Title = styled(Typography)`
  color: red;
  margin-bottom: 50px;
  margin-top: 50px;
`;

export const Title2 = styled(Typography).attrs({
  variant: 'max',
  forwardedAs: 'h3',
})`
  color: green;
`;
```


<!-- PROPS -->


