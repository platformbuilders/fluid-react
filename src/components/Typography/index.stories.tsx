import type { Meta, StoryObj } from '@storybook/react';

import Typography from './index';

const meta: Meta<typeof Typography> = {
  title: 'Typography',
  component: Typography,
  args: {
    children: 'Text',
    variant: 'xl',
  },
};

export default meta;
type Story = StoryObj<typeof Typography>;

export const Default: Story = {};
