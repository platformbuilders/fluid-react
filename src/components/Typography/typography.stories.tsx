import { expect } from '@storybook/jest';
import type { Meta, StoryObj } from '@storybook/react';
import { configure, within } from '@storybook/testing-library';

import Typography from './index';

configure({ testIdAttribute: 'id' });

//  Mocks
const mockTextId = 'typography-test-id';

const meta: Meta<typeof Typography> = {
  title: 'Components/Typography',
  component: Typography,
  parameters: {
    layout: 'centered',
  },
  tags: ['autodocs'],
  args: {
    children: 'Typography Test',
    id: mockTextId,
  },
  play: async ({ canvasElement, step }) => {
    const canvas = within(canvasElement);

    await step('Typography | Test Render', async () => {
      expect(canvas.getByTestId(mockTextId)).toBeInTheDocument();
    });
  },
};

type Story = StoryObj<typeof Typography>;

export const Default: Story = {
  args: {},
};

export default meta;
