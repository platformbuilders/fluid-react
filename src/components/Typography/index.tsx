import React, { useContext } from 'react';
import { ThemeProvider } from 'styled-components';
import { human } from 'react-native-typography';
import { Theme, Metrics, Variant } from '~/utils/types';
import { Text } from './styles';
import { ThemeContext } from '../ThemeContext';

interface Props {
  variant: Variant;
  children?: string | string[];
  style?: object[];
  textRef?: any;
  theme: Theme & Metrics;
}

const Typography: React.FC<Props> = ({
  style = [{}],
  variant = 'body',
  textRef = React.createRef(),
  children,
  ...rest
}) => {
  const { theme } = useContext(ThemeContext);
  return (
    <ThemeProvider theme={theme}>
      <Text
        ref={textRef}
        style={[human[variant], { color: theme.primary.contrast }, style]}
        {...rest}
      >
        {children}
      </Text>
    </ThemeProvider>
  );
};

export default Typography;
