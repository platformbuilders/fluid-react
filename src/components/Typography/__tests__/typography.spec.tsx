import { shallow } from 'enzyme';
import { ThemeProvider } from 'styled-components';
import { faker } from '@faker-js/faker';

import Typography from '..';
import { theme } from '../../../config/helpers';
import { TypographyType as Props } from '../../../types';

const defaultContent = 'Text';
const defaultProps: Props = {};

describe('Component: Typography', () => {
  test('snapshots', () => {
    const component = shallow(
      <ThemeProvider theme={theme}>
        <Typography {...defaultProps}>{defaultContent}</Typography>
      </ThemeProvider>,
    );

    expect(component).toMatchSnapshot();
  });

  test('check props', () => {
    // when
    const mockContent = faker.random.word();
    const component = shallow(
      <Typography {...defaultProps} variant="h1">
        {mockContent}
      </Typography>,
    );

    // then
    const { variant } = component.props();

    expect(variant).toEqual('h1');
    expect(component.children().text()).toEqual(mockContent);
  });
});
