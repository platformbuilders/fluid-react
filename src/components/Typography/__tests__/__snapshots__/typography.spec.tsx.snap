// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: Typography snapshots 1`] = `
<div>
  <p
    class="sc-bcHOTY crsAFH"
  >
    Text
  </p>
</div>
`;
